mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -95,81 +96,88 @@ template <typename T> struct MILARewritePattern : public OpRewritePattern<T> {
     return Value();
   }
 
-  Value makeSubview(PatternRewriter &b, miopen::ThreadwiseCopyOp &twcopy,
-                    Value inp) const {
-    Value subview;
-
-    auto ctx = b.getContext();
-    auto loc = inp.getLoc();
-
-    auto outputType = inp.getType().template cast<MemRefType>();
-    auto outputShape = outputType.getShape();
-    auto outputDims = outputShape.size();
-    auto outputElementType = outputType.getElementType();
-
-    // 0. initialize subview params
-    Value zero = b.create<ConstantIndexOp>(loc, 0);
-    auto zeroA = b.getIndexAttr(0);
-    SmallVector<OpFoldResult, 5> offsets(outputDims, zero);
-    SmallVector<OpFoldResult, 5> sizes(outputDims, zeroA);
-    SmallVector<OpFoldResult, 5> strides(outputDims, zeroA);
-
-    // 1. flatten input affine map with miopen::transform, (subview only allows
-    // 1 affine map)
-    auto expr = getAffineConstantExpr(0, ctx);
-    unsigned stride = 1;
-    for (int i = outputDims - 1; i >= 0; --i) {
-      expr =
-          expr + getAffineDimExpr(i, ctx) * getAffineConstantExpr(stride, ctx);
-      strides[i] = b.getIndexAttr(stride);
-      stride *= outputShape[i];
+  Value makeThreadwiseCopy(PatternRewriter &b,
+                           miopen::ThreadwiseCopyOp &miTWCopy,
+                           Value inp) const {
+    // 0. capture reg alloc for miTWCopy output regs
+    auto twinp = miTWCopy.getOperand(0);
+    auto miTransform = twinp.getDefiningOp<miopen::TransformOp>();
+    auto miTransformInp = miTransform->getOperand(0);
+    auto miAlloc = miTransformInp.getDefiningOp<miopen::GpuAllocOp>();
```

Comment:
Codes here have the assumption that the following patterns exist for `threadwise_copy` passed in:
- `gpu.alloc`
- some `miopen.transform`
- `miopen.threadwise_copy`

For logic dealing with matrix A / B in non-XDLOPS path, that would be the case. But for logic dealing with matrix C it's incorrect. We need to discuss about the use cases for this routine better.